packages/server-wallet/src/protocols/__test__/state.test.ts

```diff
@@ -0,0 +1,100 @@
+import {BN, simpleEthAllocation} from '@statechannels/wallet-core';
+
+import {alice, bob} from '../../wallet/__test__/fixtures/participants';
+import {directFundingStatus} from '../state';
+
+const zero = BN.from(0);
+const three = BN.from(3);
+const five = BN.from(5);
+const ten = BN.from(10);
+
+const outcome = simpleEthAllocation([{amount: five, destination: alice().destination}]);
+const participants = [alice(), bob()];
+
+const preFundState0 = {outcome, participants, desc: 'prefund0'};
+const preFundState1 = {outcome, participants, desc: 'prefund1'};
+const closeState = {outcome, participants, isFinal: true, desc: 'close'};
+
+const noFunding = () => ({amount: zero, transferredOut: []});
+const hasFunding5 = () => ({amount: five, transferredOut: []});
+const partTransferred = () => ({
+  amount: five,
+  transferredOut: [{toAddress: alice().destination, amount: three}],
+});
+const fullyTransferred = () => ({
+  amount: five,
+  transferredOut: [{toAddress: alice().destination, amount: five}],
+});
+
+describe('Only Alice funds channel', () =>
+  test.each`
+    supported        | fundingFn           | myParticipant      | fundingStrategy | result
+    ${preFundState0} | ${noFunding}        | ${participants[0]} | ${'Direct'}     | ${'ReadyToFund'}
```

Comment:
😻

Reply:
I'd like someone to try the alternative `test.each` api, where you call `test.each(testCases: TestCase[])`

It looks less pretty when it's static, but, it:
1. is type-safe. The type of `noFunding` is known within the test case callback.
2. is much easier to understand in git diffs

Reply:
<img width="930" alt="Screen Shot 2020-11-20 at 2 47 17 PM" src="https://user-images.githubusercontent.com/8432675/100254825-05475b00-2f00-11eb-81ef-eed35cfdb201.png">